src/lsm/tree_fuzz.zig

```diff
@@ -581,111 +585,105 @@ fn EnvironmentType(comptime table_usage: TableUsage) type {
+                }
+            }
+        }
 
-                        // Asserting the negative space:
-                        // All keys existing in our model must be checked against the scan range.
-                        if (scan_range.direction == .descending) std.mem.sort(
-                            Value,
-                            tree_values,
-                            {},
-                            struct {
-                                fn sort(_: void, a: Value, b: Value) bool {
-                                    return Table.key_from_value(&a) <
-                                        Table.key_from_value(&b);
-                                }
-                            }.sort,
-                        );
-
-                        var it = model.iterator();
-                        while (it.next()) |entry| {
-                            const model_value_key = Value.key_from_value(entry.value_ptr);
-                            const value_maybe = binary_search.binary_search_values(
-                                u64,
-                                Value,
-                                Table.key_from_value,
-                                tree_values,
-                                model_value_key,
-                                .{},
-                            );
-
-                            if (model_value_key >= scan_range.min and
-                                model_value_key <= scan_range.max)
-                            {
-                                // Must be found:
-                                if (value_maybe == null) {
-                                    // Or our buffer has exceeded, in this case the key should
-                                    // be less than the first element or greater than the last element,
-                                    // depending on the scan direction.
-                                    assert(tree_values.len == scan_results_max);
-                                    switch (scan_range.direction) {
-                                        .ascending => assert(
-                                            Table.key_from_value(&tree_values[tree_values.len - 1]) <
-                                                model_value_key,
-                                        ),
-                                        .descending => assert(
-                                            model_value_key <
-                                                Table.key_from_value(&tree_values[0]),
-                                        ),
-                                    }
-                                }
-                            } else {
-                                // Must not be found:
-                                if (value_maybe) |value| {
-                                    // Or it's a tombstone.
-                                    assert(Table.tombstone(value));
```

Comment:
I _think_ this is dead code? if the value is outside of range, it can't be found even as a tombsotne